cmd/gotk/uninstall.go

```diff
@@ -83,37 +83,55 @@ func uninstallCmdRun(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	// suspend bootstrap kustomization if it exists
+	dryRun := "--dry-run=server"
+	deleteResources := uninstallResources || uninstallCRDs
+
+	// known kinds with finalizers
+	namespacedKinds := []string{
+		sourcev1.GitRepositoryKind,
+		sourcev1.HelmRepositoryKind,
+		sourcev1.BucketKind,
+	}
+
+	// suspend bootstrap kustomization to avoid finalizers deadlock
 	kustomizationName := types.NamespacedName{
 		Namespace: namespace,
 		Name:      namespace,
 	}
 	var kustomization kustomizev1.Kustomization
-	if err := kubeClient.Get(ctx, kustomizationName, &kustomization); err == nil {
+	err = kubeClient.Get(ctx, kustomizationName, &kustomization)
+	if err == nil {
 		kustomization.Spec.Suspend = true
 		if err := kubeClient.Update(ctx, &kustomization); err != nil {
 			return fmt.Errorf("unable to suspend kustomization '%s': %w", kustomizationName.String(), err)
 		}
 	}
+	if err == nil || apierrors.IsNotFound(err) {
+		namespacedKinds = append(namespacedKinds, kustomizev1.KustomizationKind)
+	}
 
-	if uninstallResources || uninstallCRDs {
+	// suspend bootstrap Helm release if it exists
+	releaseName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      namespace,
+	}
+	var release helmv2.HelmRelease
+	err = kubeClient.Get(ctx, releaseName, &release)
+	if err == nil {
+		release.Spec.Suspend = true
+		if err := kubeClient.Update(ctx, &release); err != nil {
+			return fmt.Errorf("unable to suspend release '%s': %w", releaseName.String(), err)
+		}
+	}
+	if err == nil || apierrors.IsNotFound(err) {
+		namespacedKinds = append(namespacedKinds, helmv2.HelmReleaseKind)
+	}
```

Comment:
Given we do not bootstrap using a `HelmRelease` ourselves, this name assumption is likely wrong and the code thus obsolete?

Reply:
The code tests if the HelmRelease Kind exists, should I remove the suspend update?

Reply:
Yes, I think that would be less confusing (at least to me).